### PR TITLE
Add support for productId in transport

### PR DESCRIPTION
**Pull request by rlazo**
Optionally allow logging the productId together with the event in transport.

For googlers b/303136538

### CODE REVIEW

File: transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/internal/ExternalPRequestContext.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.google.android.datatransport.cct.internal;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class ExternalPRequestContext {
```
Comment (emilypgoogle) [inline transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/internal/ExternalPRequestContext.java:8]: `ExternalPRequestContext` is an odd name, why abbreviate just the P?

File: transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/internal/ExternalPrivacyContext.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.google.android.datatransport.cct.internal;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class ExternalPrivacyContext {
+  @Nullable
+  public abstract ExternalPRequestContext getPrequest();
```
Comment (emilypgoogle) [inline transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/internal/ExternalPrivacyContext.java:10]: This and the setter should probably both be `getPRequest` if we're sticking with that name.

**Review by emilypgoogle — APPROVED** [review]
Just a naming crit on PRequest, otherwise looks good

File: transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/internal/ExternalPRequestContext.java (hunk 1 above)
Comment (rlazo) [reply transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/internal/ExternalPRequestContext.java:22]: Discussed offline

File: transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/internal/ExternalPrivacyContext.java (hunk 2 above)
Comment (rlazo) [reply transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/internal/ExternalPrivacyContext.java:24]: Discussed offline

**Pull request merged**